### PR TITLE
NT-1776:Refactor SignupActivity

### PR DESCRIPTION
**Pull request by hadia**
# 📲 What

Remove butter knife from SignupActivity and replaced with view binding

# 🤔 Why

deprecated butterknife

# 🛠 How

1- change SignupActivity from java to kotlin
3-remove butter-knife and replace it with viewbinding

# 👀 See

https://user-images.githubusercontent.com/1075310/108105578-79c00f00-7095-11eb-9368-3b8aad2eab4e.mp4
| Before 🐛 | After 🦋 |
| --- | --- |
|  |  |

# 📋 QA
Open app try to sign up
# Story 📖

https://kickstarter.atlassian.net/browse/NT-1776

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ui/activities/SignupActivity.kt
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/activities/SignupActivity.kt:124]: this if condition can be deleted, it was mean to not allow pressing back button on the `Interstitial Fragment` that screen is no longer presented

File: app/src/main/java/com/kickstarter/ui/activities/SignupActivity.kt
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/activities/SignupActivity.kt:123]: This override method can be deleted, it was meant to avoid pressing the back button on the `Interstitial Fragment` that is no longer used

**Review by Arkariang — CHANGES_REQUESTED** [review]
Apart from the override method everything else looks good, I've tested the signUp with errors scenarios and all works fine. As soon as that override method is deleted I'll approve it 
SignUpErrorCase

File: app/src/main/java/com/kickstarter/ui/activities/SignupActivity.kt
Comment (leighdouglas) [inline app/src/main/java/com/kickstarter/ui/activities/SignupActivity.kt:28]: nit: whitespace after line 27

**Review by leighdouglas — APPROVED** [review]
LGTM once @Arkariang approves 🎉

**Review by Arkariang — APPROVED** [review]
Lgtm!

**Pull request merged**
